NPE in PgpKey.encryptKey()

### Where? 
The key encryption

### Need to look at

```
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.bouncycastle.bcpg.BCPGKey org.bouncycastle.openpgp.PGPPrivateKey.getPrivateKeyDataPacket()' on a null object reference
        at org.bouncycastle.openpgp.PGPSecretKey.buildSecretKeyPacket(Unknown Source:6)
        at org.bouncycastle.openpgp.PGPSecretKey.<init>(Unknown Source:5)
```
### Full stack
```
[SUMMARY]
java.lang.NullPointerException: Attempt to invoke virtual method 'org.bouncycastle.bcpg.BCPGKey org.bouncycastle.openpgp.PGPPrivateKey.getPrivateKeyDataPacket()' on a null object reference
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:


[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:30)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:60)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.bouncycastle.bcpg.BCPGKey org.bouncycastle.openpgp.PGPPrivateKey.getPrivateKeyDataPacket()' on a null object reference
        at org.bouncycastle.openpgp.PGPSecretKey.buildSecretKeyPacket(Unknown Source:6)
        at org.bouncycastle.openpgp.PGPSecretKey.<init>(Unknown Source:5)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.lockPrivateKey(SecretKeyRingEditor.java:703)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.changePassphrase(SecretKeyRingEditor.java:675)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.access$300(SecretKeyRingEditor.java:71)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor$WithPassphraseImpl.toNewPassphrase(SecretKeyRingEditor.java:647)
        at com.flowcrypt.email.example.file.PgpKey.encryptKey(PgpKey.kt:38)
        at com.flowcrypt.email.example.file.PgpKey.encryptKey(PgpKey.kt:31)
        at com.flowcrypt.email.service.actionqueue.actions.EncryptPrivateKeysIfNeededAction.run(EncryptPrivateKeysIfNeededAction.kt:71)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:54)
        ... 7 more
java.lang.NullPointerException: Attempt to invoke virtual method 'org.bouncycastle.bcpg.BCPGKey org.bouncycastle.openpgp.PGPPrivateKey.getPrivateKeyDataPacket()' on a null object reference
        at org.bouncycastle.openpgp.PGPSecretKey.buildSecretKeyPacket(Unknown Source:6)
        at org.bouncycastle.openpgp.PGPSecretKey.<init>(Unknown Source:5)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.lockPrivateKey(SecretKeyRingEditor.java:703)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.changePassphrase(SecretKeyRingEditor.java:675)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor.access$300(SecretKeyRingEditor.java:71)
        at org.example.file.modification.secretkeyring.SecretKeyRingEditor$WithPassphraseImpl.toNewPassphrase(SecretKeyRingEditor.java:647)
        at com.flowcrypt.email.example.file.PgpKey.encryptKey(PgpKey.kt:38)
        at com.flowcrypt.email.example.file.PgpKey.encryptKey(PgpKey.kt:31)
        at com.flowcrypt.email.service.actionqueue.actions.EncryptPrivateKeysIfNeededAction.run(EncryptPrivateKeysIfNeededAction.kt:71)
        at com.flowcrypt.email.service.actionqueue.ActionQueueIntentService.onHandleWork(ActionQueueIntentService.kt:54)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:392)
        at androidx.core.app.JobIntentService$CommandProcessor.doInBackground(JobIntentService.java:383)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt
@@ -5,25 +5,14 @@
 
 package com.flowcrypt.email.security.pgp
 
-import com.flowcrypt.email.api.retrofit.response.model.node.MsgBlock
 import com.flowcrypt.email.extensions.pgp.armor
 import com.flowcrypt.email.extensions.pgp.toNodeKeyDetails
-import org.bouncycastle.bcpg.ArmoredInputStream
 import org.bouncycastle.openpgp.PGPKeyRing
-import org.bouncycastle.openpgp.PGPObjectFactory
-import org.bouncycastle.openpgp.PGPPublicKey
-import org.bouncycastle.openpgp.PGPPublicKeyRing
-import org.bouncycastle.openpgp.PGPSecretKey
 import org.bouncycastle.openpgp.PGPSecretKeyRing
-import org.bouncycastle.openpgp.PGPSignature
-import org.bouncycastle.openpgp.jcajce.JcaPGPPublicKeyRingCollection
-import org.bouncycastle.openpgp.jcajce.JcaPGPSecretKeyRingCollection
-import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator
 import org.pgpainless.PGPainless
 import org.pgpainless.key.collection.PGPKeyRingCollection
 import org.pgpainless.util.Passphrase
 import java.io.InputStream
-import java.nio.charset.StandardCharsets
 
 @Suppress("unused")
 object PgpKey {
@@ -102,102 +91,10 @@ object PgpKey {
     if (parseKeyResult.getAllKeys().isEmpty()) {
       throw IllegalArgumentException("Keys not found")
     }
-    if (parseKeyResult.pgpKeyRingCollection.pgpSecretKeyRingCollection.first() is PGPSecretKeyRing)
+    if (parseKeyResult.pgpKeyRingCollection.pgpSecretKeyRingCollection.firstOrNull() is PGPSecretKeyRing)
       return parseKeyResult.pgpKeyRingCollection.pgpSecretKeyRingCollection.first()
     else
       throw IllegalArgumentException("Key is not a secret key")
-  }
-
-  private fun parseAndNormalizeKeyRings(armored: String): List<PGPKeyRing> {
-    val normalizedArmored = PgpArmor.normalize(armored, MsgBlock.Type.UNKNOWN)
-    val keys = mutableListOf<PGPKeyRing>()
-    if (PgpArmor.ARMOR_HEADER_DICT_REGEX[MsgBlock.Type.PUBLIC_KEY]!!
-            .beginRegexp.containsMatchIn(normalizedArmored)) {
-      val keyRingCollection = JcaPGPPublicKeyRingCollection(
-          ArmoredInputStream(normalizedArmored.toByteArray(StandardCharsets.UTF_8).inputStream())
-      )
-      // We have to use reflection because BouncyCastle declares "order" list as a private field
-      // https://stackoverflow.com/a/1196207/1540501
-      // Sent a request to BouncyCastle mailing list to make this possible in a better way.
-      val orderField = keyRingCollection.javaClass.superclass!!.getDeclaredField("order")
-      orderField.isAccessible = true
-      keys.addAll(
-          (orderField.get(keyRingCollection) as java.util.List<java.lang.Long>).map {
-            keyRingCollection.getPublicKeyRing(it.toLong())!!
-          }
-      )
-    } else if (PgpArmor.ARMOR_HEADER_DICT_REGEX[MsgBlock.Type.PRIVATE_KEY]!!
-            .beginRegexp.containsMatchIn(normalizedArmored)) {
-      val keyRingCollection = JcaPGPSecretKeyRingCollection(
-          ArmoredInputStream(normalizedArmored.toByteArray(StandardCharsets.UTF_8).inputStream())
-      )
-      // Again, use reflection because BouncyCastle declares "order" list as a private field.
-      val orderField = keyRingCollection.javaClass.superclass!!.getDeclaredField("order")
-      orderField.isAccessible = true
-      keys.addAll(
-          (orderField.get(keyRingCollection) as java.util.List<java.lang.Long>).map {
-            keyRingCollection.getSecretKeyRing(it.toLong())!!
-          }
-      )
-    } else if (PgpArmor.ARMOR_HEADER_DICT_REGEX[MsgBlock.Type.ENCRYPTED_MSG]!!
-            .beginRegexp.containsMatchIn(normalizedArmored)) {
-      val objectFactory = PGPObjectFactory(
-          ArmoredInputStream(normalizedArmored.toByteArray(StandardCharsets.UTF_8).inputStream()),
-          JcaKeyFingerprintCalculator()
-      )
-      while (true) {
-        val obj = objectFactory.nextObject() ?: break
-        if (obj is PGPKeyRing) {
-          keys.add(obj)
-        }
-      }
-    }
-
-    // Prevent key bloat by removing all non-self certifications
-    for ((keyRingIndex, keyRing) in keys.withIndex()) {
-      val primaryKeyID = keyRing.publicKey.keyID
-      if (keyRing is PGPPublicKeyRing) {
-        var replacementKeyRing: PGPPublicKeyRing = keyRing
-        for (publicKey in keyRing.publicKeys) {
-          var replacementKey = publicKey
-          for (sig in publicKey.signatures.asSequence().map { it as PGPSignature }.filter {
-            it.isCertification && it.keyID != primaryKeyID
-          }) {
-            replacementKey = PGPPublicKey.removeCertification(replacementKey, sig)
-          }
-          if (replacementKey !== publicKey) {
-            replacementKeyRing = PGPPublicKeyRing.insertPublicKey(
-                replacementKeyRing, replacementKey
-            )
-          }
-        }
-        if (replacementKeyRing !== keyRing) {
-          keys[keyRingIndex] = replacementKeyRing
-        }
-      } else if (keyRing is PGPSecretKeyRing) {
-        var replacementKeyRing: PGPSecretKeyRing = keyRing
-        for (secretKey in keyRing.secretKeys) {
-          val publicKey = secretKey.publicKey
-          var replacementPublicKey = publicKey
-          for (sig in publicKey.signatures.asSequence().map { it as PGPSignature }.filter {
-            it.isCertification && it.keyID != primaryKeyID
-          }) {
-            replacementPublicKey = PGPPublicKey.removeCertification(replacementPublicKey, sig)
-          }
-          if (replacementPublicKey !== publicKey) {
-            val replacementKey = PGPSecretKey.replacePublicKey(secretKey, replacementPublicKey)
-            replacementKeyRing = PGPSecretKeyRing.insertSecretKey(
-                replacementKeyRing, replacementKey
-            )
-          }
-        }
-        if (replacementKeyRing !== keyRing) {
-          keys[keyRingIndex] = replacementKeyRing
-        }
-      }
-    }
-
-    return keys
   }
 
   data class ParseKeyResult(val pgpKeyRingCollection: PGPKeyRingCollection) {

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpKey.kt
@@ -27,53 +27,31 @@ import java.nio.charset.StandardCharsets
 
 @Suppress("unused")
 object PgpKey {
+  /**
+   * Encrypt the given key.
+   *
+   * @param armored Should be a single private key.
+   */
   fun encryptKey(armored: String, passphrase: String): String {
-    return encryptKey(extractSecretKey(armored), passphrase).armor()
+    return encryptKey(extractSecretKeyRing(armored), passphrase).armor()
   }
 
-  private fun encryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
-    return PGPainless.modifyKeyRing(key)
-        .changePassphraseFromOldPassphrase(null)
-        .withSecureDefaultSettings()
-        .toNewPassphrase(Passphrase.fromPassword(passphrase))
-        .done()
-  }
-
+  /**
+   * Decrypt the given key.
+   *
+   * @param armored Should be a single private key.
+   */
   fun decryptKey(armored: String, passphrase: String): String {
-    return decryptKey(extractSecretKey(armored), passphrase).armor()
+    return decryptKey(extractSecretKeyRing(armored), passphrase).armor()
   }
 
-  private fun decryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
-    return PGPainless.modifyKeyRing(key)
-        .changePassphraseFromOldPassphrase(Passphrase.fromPassword(passphrase))
-        .withSecureDefaultSettings()
-        .toNoPassphrase()
-        .done()
-  }
-
-  fun changeKeyPassphrase(
-      armored: String,
-      oldPassphrase: String,
-      newPassphrase: String
-  ): String {
-    return changeKeyPassphrase(extractSecretKey(armored), oldPassphrase, newPassphrase).armor()
-  }
-
-  private fun changeKeyPassphrase(
-      key: PGPSecretKeyRing,
-      oldPassphrase: String,
-      newPassphrase: String
-  ): PGPSecretKeyRing {
-    return encryptKey(decryptKey(key, oldPassphrase), newPassphrase)
-  }
-
-  private fun extractSecretKey(armored: String): PGPSecretKeyRing {
-    val key = parseAndNormalizeKeyRings(armored).firstOrNull()
-        ?: throw IllegalArgumentException("Keys not found")
-    if (key is PGPSecretKeyRing)
-      return key
-    else
-      throw IllegalArgumentException("Key is not a secret key")
+  /**
+   * Change a passphrase for the given key.
+   *
+   * @param armored Should be a single private key.
+   */
+  fun changeKeyPassphrase(armored: String, oldPassphrase: String, newPassphrase: String): String {
+    return changeKeyPassphrase(extractSecretKeyRing(armored), oldPassphrase, newPassphrase).armor()
   }
 
   fun parseKeys(source: String, throwExceptionIfUnknownSource: Boolean = true): ParseKeyResult {
@@ -93,6 +71,41 @@ object PgpKey {
    */
   fun parseKeys(source: InputStream, throwExceptionIfUnknownSource: Boolean = true): ParseKeyResult {
     return ParseKeyResult(PGPainless.readKeyRing().keyRingCollection(source, throwExceptionIfUnknownSource))
+  }
+
+  private fun decryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
+    return PGPainless.modifyKeyRing(key)
+        .changePassphraseFromOldPassphrase(Passphrase.fromPassword(passphrase))
+        .withSecureDefaultSettings()
+        .toNoPassphrase()
+        .done()
+  }
+
+  private fun encryptKey(key: PGPSecretKeyRing, passphrase: String): PGPSecretKeyRing {
+    return PGPainless.modifyKeyRing(key)
+        .changePassphraseFromOldPassphrase(null)
+        .withSecureDefaultSettings()
+        .toNewPassphrase(Passphrase.fromPassword(passphrase))
+        .done()
+  }
+
+  private fun changeKeyPassphrase(
+      key: PGPSecretKeyRing,
+      oldPassphrase: String,
+      newPassphrase: String
+  ): PGPSecretKeyRing {
+    return encryptKey(decryptKey(key, oldPassphrase), newPassphrase)
+  }
+
+  private fun extractSecretKeyRing(armored: String): PGPSecretKeyRing {
+    val parseKeyResult = parseKeys(armored)
+    if (parseKeyResult.getAllKeys().isEmpty()) {
+      throw IllegalArgumentException("Keys not found")
+    }
+    if (parseKeyResult.pgpKeyRingCollection.pgpSecretKeyRingCollection.first() is PGPSecretKeyRing)
+      return parseKeyResult.pgpKeyRingCollection.pgpSecretKeyRingCollection.first()
+    else
+      throw IllegalArgumentException("Key is not a secret key")
   }
 
   private fun parseAndNormalizeKeyRings(armored: String): List<PGPKeyRing> {

